frontend/packages/ceph-storage-plugin/src/components/ocs-install/create-ocs-service.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { match } from 'react-router';
+import {
+  K8sResourceKind,
+  K8sKind,
+  k8sGet,
+  K8sResourceKindReference,
+} from '@console/internal/module/k8s';
+import { BreadCrumbs } from '@console/internal/components/utils/index';
+import { ClusterServiceVersionModel } from '@console/internal/models';
+import { ClusterServiceVersionKind } from '@console/internal/components/operator-lifecycle-manager/index';
+import { OCSServiceModel } from '../../models';
+import { CreateOCSServiceForm } from './create-form';
+import { CreateOCSServiceYAML } from './create-yaml';
+
+/**
+ * Component which wraps the YAML editor and form together
+ */
+export const CreateOCSService: React.FC<CreateOCSServiceProps> = React.memo((props) => {
+  const [sample, setSample] = React.useState(null);
+  const [method, setMethod] = React.useState<'yaml' | 'form'>('form');
+  const [clusterServiceVersion, setClusterServiceVersion] = React.useState(null);
+
+  React.useEffect(() => {
+    k8sGet(ClusterServiceVersionModel, props.match.params.appName, props.match.params.ns)
+      .then((clusterServiceVersionObj) => {
+        try {
+          setSample(
+            JSON.parse(_.get(clusterServiceVersionObj.metadata.annotations, 'alm-examples'))[0],
+          );
+          setClusterServiceVersion(clusterServiceVersionObj);
+        } catch (e) {
+          setClusterServiceVersion(null);
+        }
+      })
+      .catch(() => setClusterServiceVersion(null));
+  }, [props.match.params.appName, props.match.params.ns]);
+
+  return (
+    <React.Fragment>
+      <div className="co-create-operand__header">
+        <div className="co-create-operand__header-buttons">
+          {clusterServiceVersion !== null && (
+            <BreadCrumbs
+              breadcrumbs={[
+                {
+                  name: clusterServiceVersion.spec.displayName,
+                  path: window.location.pathname.replace('/~new', ''),
```

Comment:
I don't think this handles custom context roots correctly. You should be using `props.match.url` from react-router

https://reacttraining.com/react-router/web/api/match

```suggestion
                  path: props.match.url.replace('/~new', ''),
```

Reply:
ack